### ISSUE TITLE
Could not bind app to MongoDB Atlas PCF service instance

### ISSUE THREAD

**Issue by alex-tw-lam** [mongodb/atlas-osb#53]
### Reproducing steps

```
cf target -o <PCF ORG NAME> -s <PCF SPACE NAME>
cf bind-service <PCF APP NAME> <PCF SERVICE INSTANCE NAME>
```

### Error message

```
Service broker error: cannot find plan for instance "c20bdcea-6bf1-4804-ab06-bed618054e7d": cannot fetch instance: cannot find instance in maintenance DB(s): no instances found
```

**Comment by alex-tw-lam** [comment]
Noticed a similar issue here https://github.com/mongodb/atlas-osb/issues/27#issuecomment-720421545

Given 0.7.0 and 0.5.0 are incompatible, we need to figure ways to handle instances created with older broker version. 

#### Questions:
1. Is  _purge and re-create_ the only work-around? Is there less intrusive ways? 
2. How do we list out instances created with 0.5.0?

**Comment by alanlunspace** [comment]
Duplicated https://github.com/mongodb/atlas-osb/issues/51.

**Comment by vasilevp** [comment]
Steps to repair the affected environments:

1. Upgrade to Broker v0.8.0 or later. Note that v0.8.0+ detects broken state and will refuse to work until you complete step 2
2. Run the [migration tool](vasilevp/migration-tool) against your `Atlas Service Broker Mainentance` [sic] project
   1. Create a Project-level API Key with `Project Owner` permission or use existing one
   2. Pass the Project ID (`https://cloud.mongodb.com/v2/ --> project_id <-- #clusters`), Public API Key (`abcdefgh`) and Private API Key (`12345678-1234-1234-1234-12345678abcd`) to the tool:
     `./migration-tool --publickey abcdefgh --privatekey 12345678-1234-1234-1234-12345678abcd --projectid abcdef1234567890`
3. Verify that migration was successful:
   - `migration-tool` did not return any errors
   - `Atlas Service Broker Mainentance` has only one Realm App called `broker-state`
   - The Broker is not complaining about multiple `broker-state` apps
4. Delete `backup.*.json`

**Comment by vasilevp** [comment]
Closing this one since the fix was shipped with v0.8.0. Please open a new issue if you encounter problems with migration, or a similar issue occurs again in a newer version of the Broker.
